packages/core/lib/attributes.ts

```diff
@@ -53,13 +53,54 @@ export type ResourceAttributeSource<C extends Configuration>
 
 export interface JsonAttribute {
   key: string
-  value: { stringValue: string }
-  | { intValue: string }
-  | { doubleValue: number }
-  | { boolValue: boolean }
+  value: { stringValue: string } | { intValue: string } | { doubleValue: number } | { boolValue: boolean }
 }
 
-export function attributeToJson (key: string, attribute: SpanAttribute): JsonAttribute | undefined {
+type AttributeArrayValueValue = Array<{ stringValue: string } | { intValue: string } | { doubleValue: number } | { boolValue: boolean }>
```

Comment:
Should we be enforcing a single-type array here? maybe something like this?

```suggestion
type AttributeArrayValueValue = Array<{ stringValue: string }> | Array<{ intValue: string }> | Array<{ doubleValue: number }> | Array<{ boolValue: boolean }>
```

Reply:
actually no! those are perfectly valid - but when we _set_ them we try to enforce a single type - as per the `SpanAttribute` above

Reply:
ah yeah of course - this is the internal type for the resulting JSON 🤦🏾

Reply:
I think I need to update the function and type names to make things a bit clearer

Reply:
made some updates - is this a little easier to read?

Reply:
yeah that's a lot clearer - how about going a step further and making an `ArrayAttributeValue` interface?

```ts
interface StringAttributeValue { stringValue: string }
interface IntAttributeValue { intValue: string }
interface DoubleAttributeValue { doubleValue: number }
interface BoolAttributeValue { boolValue: boolean }

type SingleAttributeValue = StringAttributeValue | IntAttributeValue | DoubleAttributeValue | BoolAttributeValue

interface ArrayAttributeValue {
  arrayValue: {
    values?: SingleAttributeValue[]
  }
}

type JsonAttributeValue = SingleAttributeValue | ArrayAttributeValue

export interface JsonAttribute {
  key: string
  value: JsonAttributeValue
}

```

Reply:
Done and done!